File: android-lib/src/androidTest/java/com/kinvey/androidTest/model/PersonList.java
package com.kinvey.androidTest.model;

import com.google.api.client.util.Key;

import java.util.List;

public class PersonList extends Person{

    @Key("list")
    private List<PersonList> list;

    public PersonList() {
    }

    public PersonList(String name) {
        this.username = name;
    }

    public List<PersonList> getList() {
        return list;
    }

    public void setList(List<PersonList> list) {
        this.list = list;
    }
}

File: android-lib/src/androidTest/java/com/kinvey/androidTest/model/PersonList.kt
package com.kinvey.androidTest.model

import com.google.api.client.util.Key

class PersonList(
    @Key("list")
    var list: List<PersonList>? = null,
    @Key("personList")
    var personList: PersonList? = null
) : Person() {
    constructor(name: String?): this() {
        username = name
    }
}